### PR TITLE
[disasters] flask endpoint updates

### PR DESCRIPTION
**Pull request by chejennifer**
- take a date range instead of a single date when fetching event data
- filter out values that are missing the unit for severity

### CODE REVIEW

File: static/js/utils/disaster_event_map_utils.tsx [hunk 1]
```diff
@@ -339,55 +346,27 @@ function getDateRange(selectedDate: string): [string, string] {
...
+      promiseSpecId.push(eventSpec.id);
+      promises.push(
+        fetchEventPoints(
+          eventType,
+          dataOptions.place,
+          dateRange,
+          eventSpec.id,
+          dataOptions.severityFilters[eventSpec.id],
+          dateRange[0].length > 7 ? dateRange[0] : "",
+          dateRange[1].length > 7 ? dateRange[1] : "",
```
Comment (beets) [inline static/js/utils/disaster_event_map_utils.tsx:366]: i don't understand the default value here. if dateRange is a year, we should still honor it. how about leaving that handling to the endpoint?

**Review by beets — APPROVED** [review]
sorry for the late review! i missed this in the queue

File: server/routes/api/disaster_api.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
 EARTH_DCID = "Earth"
 EVENT_POINT_KEYS = set(
     ["affectedPlaces", "latitude", "longitude", "startDate", "eventId"])
+DATA_RETRIEVAL_DATE_LENGTH = 7
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply server/routes/api/disaster_api.py:28]: done

File: server/routes/api/disaster_api.py [hunk 3]
```diff
@@ -71,20 +72,42 @@ def keep_event(event, place, filter_prop, filter_unit, filter_upper_limit,
...
     )
-    return True
+    return False
+
+
+def get_date_list(min_date: str, max_date: str):
+  """
+  Given a date range, gets the list of dates to retrieve data for.
+  min_date and max_date must have the same date format of YYYY or YYYY-MM or
+  YYYY-MM-DD.
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply server/routes/api/disaster_api.py]: done

File: static/js/utils/disaster_event_map_utils.tsx (hunk 1 above)
Comment (chejennifer) [reply static/js/utils/disaster_event_map_utils.tsx]: ah, good catch, this was from old logic, removed it

**Comment by chejennifer** [comment]
thanks for the review!

**Pull request merged**